src/util/std_expr.h

```diff
@@ -3194,10 +3165,9 @@ inline address_of_exprt &to_address_of_expr(exprt &expr)
 class not_exprt:public unary_exprt
 {
 public:
-  explicit not_exprt(const exprt &op):
-    unary_exprt(ID_not, op) // type from op.type()
+  explicit not_exprt(exprt _op) : unary_exprt(ID_not, std::move(_op))
   {
-    PRECONDITION(op.type().id()==ID_bool);
+    PRECONDITION(op().type().id() == ID_bool);
   }
```

Comment:
Good catch

Reply:
Hmm, but then I should really add `as_const()` in there ;-)

Reply:
Unfortunately, this doesn't do the job -- the mere access to `operands()` is enough to wreck the sharing.
This needs to be `as_const(*this).type().id()`.

Reply:
My apologies for failing to take into account this remark right here, the fix is now in #4391.